lib/streamlit/elements/widgets/selectbox.py

```diff
@@ -51,41 +52,80 @@
 @dataclass
 class SelectboxSerde(Generic[T]):
     options: Sequence[T]
-    index: int
+    index: int | None
 
-    def serialize(self, v: object) -> int:
+    def serialize(self, v: object) -> int | None:
+        if v is None:
+            return None
         if len(self.options) == 0:
             return 0
         return index_(self.options, v)
 
     def deserialize(
         self,
-        ui_value: Optional[int],
+        ui_value: int | None,
         widget_id: str = "",
-    ) -> Optional[T]:
-        idx: int = ui_value if ui_value is not None else self.index
-
-        return self.options[idx] if len(self.options) > 0 else None
+    ) -> T | None:
+        idx = ui_value if ui_value is not None else self.index
+        return self.options[idx] if idx is not None and len(self.options) > 0 else None
 
 
 class SelectboxMixin:
-    @gather_metrics("selectbox")
+    @overload
     def selectbox(
         self,
         label: str,
         options: OptionSequence[T],
         index: int = 0,
         format_func: Callable[[Any], Any] = str,
-        key: Optional[Key] = None,
-        help: Optional[str] = None,
-        on_change: Optional[WidgetCallback] = None,
-        args: Optional[WidgetArgs] = None,
-        kwargs: Optional[WidgetKwargs] = None,
+        key: Key | None = None,
+        help: str | None = None,
+        on_change: WidgetCallback | None = None,
+        args: WidgetArgs | None = None,
+        kwargs: WidgetKwargs | None = None,
+        *,  # keyword-only arguments:
+        placeholder: str = "Select...",
+        disabled: bool = False,
+        label_visibility: LabelVisibility = "visible",
+    ) -> T:
+        pass
+
+    @overload
+    def selectbox(
+        self,
+        label: str,
+        options: OptionSequence[T],
+        index: None = None,
+        format_func: Callable[[Any], Any] = str,
+        key: Key | None = None,
+        help: str | None = None,
+        on_change: WidgetCallback | None = None,
+        args: WidgetArgs | None = None,
+        kwargs: WidgetKwargs | None = None,
         *,  # keyword-only arguments:
         placeholder: str = "Select...",
         disabled: bool = False,
         label_visibility: LabelVisibility = "visible",
-    ) -> Optional[T]:
+    ) -> T | None:
+        pass
```

Comment:
I think my comments on #7269 should also apply here:
* https://github.com/streamlit/streamlit/pull/7269#discussion_r1316611057
* https://github.com/streamlit/streamlit/pull/7269#discussion_r1316611561

Reply:
Removed the overloads for `st.radio` and `st.selectbox` as mentioned in the comment in `st.radio`.